### PR TITLE
add more type on codeigniter custom install

### PR DESCRIPTION
**Pull request by sukrosono**

### CODE REVIEW

**Review by niksamokhvalov — CHANGES_REQUESTED** [review]
Add tests, please.

**Comment by sukrosono** [comment]
Oh yeah, i also want more type. Would you point me the basic test case?

**Comment by niksamokhvalov** [comment]
You need to extend basic test case in `tests/Composer/Installers/Test/InstallerTest.php`.

**Comment by sukrosono** [comment]
Yeah, basically I just copy-paste the test case from cake.
this is my first unit test on PHP, It the same as mocking?

**Comment by sukrosono** [comment]
Could you make a minor release ASAP after this one merged, please?

File: src/Composer/Installers/CodeIgniterInstaller.php
Comment (niksamokhvalov) [inline src/Composer/Installers/CodeIgniterInstaller.php:19]: Hmm... Why used non-existed version? I don't see this version on releases page: https://github.com/bcit-ci/CodeIgniter/releases.

**Review by niksamokhvalov — COMMENTED** [review]
1. I don't understand these changes. It's PR include not only new type for Codeigniter, but and changes with additional logic. Can you explain these changes in detail?
1. Tests in CI is broken.

Sorry, I steel don't merge it.

**Comment by sukrosono** [comment]
1. You can suggest title changes, in your point of view would be more relevant.
2. Yes, I am not surprised with PHP 5 Travis failures, I will take a look, but personally I would drop PHP 5 support.

File: src/Composer/Installers/CodeIgniterInstaller.php
Comment (sukrosono) [reply src/Composer/Installers/CodeIgniterInstaller.php:19]: This is the why
image

**Comment by sukrosono** [comment]
I need your approval to run the Travis build, I am already trying to use my own Travis, but I am giving in to do that options.
If you have some suggestion, let me know.
My future work will depend on this.

File: src/Composer/Installers/CodeIgniterInstaller.php
Comment (niksamokhvalov) [reply src/Composer/Installers/CodeIgniterInstaller.php]: The current version is 3.1.11, but you comparing with 3.11 version. It's error?

**Review by niksamokhvalov — CHANGES_REQUESTED** [review]
1. Tests in CI is broken.
2. Compare `CodeIgniterInstaller::$locations` with types description in `README.md`. They are different.

**Comment by sukrosono** [comment]
That's awkward, I don't have free time yet to continue. The failure is not on all PHP versions, just likely to fail on 5.3. I can't dig deeper into the CI, I just rely on manual tests before asking you to merge.

**Comment by Seldaek** [comment]
Closing as this is so old it is probably irrelevant at this point. If not let me know.

**Pull request closed without merging**
